numpyro/contrib/funsor/enum_messenger.py

```diff
@@ -480,7 +480,7 @@ def process_message(self, msg):  # copied almost verbatim from plate
                 msg['kwargs']['sample_shape'] = ()
             overlap_idx = max(len(expected_shape) - len(dist_batch_shape), 0)
             trailing_shape = expected_shape[overlap_idx:]
-            broadcast_shape = lax.broadcast_shapes(trailing_shape, dist_batch_shape)
+            broadcast_shape = lax.broadcast_shapes(trailing_shape, tuple(dist_batch_shape))
```

Comment:
I cast batch_shape to `tuple` because without doing so, this will raise `TypeError: unhashable type: 'TensorShape'`
```python
from numpyro.contrib.tfp import distributions as tfd
from jax.lax import broadcast_shapes
d = tfd.Gamma(1, 1)
broadcast_shapes(d.batch_shape, d.event_shape)
```